system/mp/mpcomm.cpp

```diff
@@ -2237,6 +2237,37 @@ int CMPConnectThread::run()
                 connectHdr.id[0].get(_remoteep);
                 connectHdr.id[1].get(hostep);
 
+                // HPCC-28125 - if remoteep IP != peerEp IP then log warning - do we change remoteep ?
+                //              if hostep != listen ep then log warning - do we change hostep ?
+                // **VERY CONCERNED** about pods and local IPs and Network Address Translation
+
+                if (!_remoteep.ipequals(peerEp))
+                {
+                    StringBuffer rIp, pIp;
+                    _remoteep.getIpText(rIp);
+                    peerEp.getIpText(pIp);
+                    PROGLOG("MP: WARNING: remote client address (%s:%u) != peer address (%s:%u)", rIp.str(), _remoteep.port, pIp.str(), peerEp.port);
+                    _remoteep.set(peerEp);
+                    connectHdr.id[0].set(_remoteep);
+                    // should we sock->close() and continue ?
+                }
+
+                SocketEndpoint listenEp;
+                listensock->getEndpoint(listenEp);
+                unsigned short lPort = listenEp.port;
```

Comment:
this is in order to get thte port of the listening socket..
rather than call getEndpoint which entails a system call etc., it would probably be better to store the port in the constructor as a member and reuse it here.

Reply:
Yes, I can look into that.